projects/models.py

```diff
@@ -78,6 +92,97 @@ class ProjectExtras(models.Model):
     nickname = models.CharField(max_length=50, blank=False, null=False, unique=True)
 
 
+class InvitationQuerySet(models.QuerySet):
+    pass
+
+
+class InvitationManager(models.Manager):
+    pass
+
+
+class Invitation(models.Model):
+    """Model to hold invitations of users to projects."""
+
+    STATUS = [("ISSUED", "Issued"), ("ACCEPTED", "Accepted")]
+
+    @staticmethod
+    def default_days_until_expiration():
+        return 30
+
+    def _generate_secret():
+        """Generate secure code.
+
+        https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe
+        Each byte creates 1.3 characters on average, but we need to store into
+        a fixed length field. Using nchars for both bytes and characters ensures
+        that we always have sufficient randomness, but will fit in the db field.
+        As of 2021, a resonable standard is 20 bytes of randomness, or 26 characters.
+        """
+        nbytes = 26
+        nchars = nbytes
+        return secrets.token_urlsafe(nbytes)[:nchars]
+
+    def _generate_expiration():
+        now = timezone.now()
+        duration = timezone.timedelta(days=Invitation.default_days_until_expiration())
+        return now + duration
+
+    # This information is needed on creation
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    user_issued = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.DO_NOTHING, editable=True
```

Comment:
What does `editable` mean here? Is that something for the admin UI? If so, why is it editable?

Reply:
`editable` is for the admin UI, and so it is not needed here.